src/apache/conf/httpd.conf

```diff
@@ -193,3 +193,37 @@ ServerSignature Off
 <IfDefine EnableHTTPS>
     Include ${SNAP}/conf/ssl.conf
 </IfDefine>
+
+<IfDefine EnablePrettylink>
+    # From here: https://github.com/nextcloud/server/blob/stable21/lib/private/Setup.php#L523-L561
+    ErrorDocument 403 ${WEBROOT}/
+    ErrorDocument 404 ${WEBROOT}/
+    <IfModule mod_rewrite.c>
+    Options -MultiViews
+    RewriteRule ^core/js/oc.js$ index.php [PT,E=PATH_INFO:]
+    RewriteRule ^core/preview.png$ index.php [PT,E=PATH_INFO:]
+    RewriteCond %{REQUEST_FILENAME} !\.(css|js|svg|gif|png|html|ttf|woff2?|ico|jpg|jpeg|map|webm|mp4|mp3|ogg|wav)$
+    RewriteCond %{REQUEST_FILENAME} !core/img/favicon.ico$
+    RewriteCond %{REQUEST_FILENAME} !core/img/manifest.json$
+    RewriteCond %{REQUEST_FILENAME} !/remote.php
+    RewriteCond %{REQUEST_FILENAME} !/public.php
+    RewriteCond %{REQUEST_FILENAME} !/cron.php
+    RewriteCond %{REQUEST_FILENAME} !/core/ajax/update.php
+    RewriteCond %{REQUEST_FILENAME} !/status.php
+    RewriteCond %{REQUEST_FILENAME} !/ocs/v1.php
+    RewriteCond %{REQUEST_FILENAME} !/ocs/v2.php
+    RewriteCond %{REQUEST_FILENAME} !/robots.txt
+    RewriteCond %{REQUEST_FILENAME} !/updater/
+    RewriteCond %{REQUEST_FILENAME} !/ocs-provider/
+    RewriteCond %{REQUEST_FILENAME} !/ocm-provider/
+    RewriteCond %{REQUEST_URI} !^/\.well-known/(acme-challenge|pki-validation)/.*
+    RewriteCond %{REQUEST_FILENAME} !/richdocumentscode(_arm64)?/proxy.php$
+    RewriteRule . index.php [PT,E=PATH_INFO:]
+    RewriteBase ${REWRITEBASE}
+    <IfModule mod_env.c>
+        SetEnv front_controller_active true
+        <IfModule mod_dir.c>
+        DirectorySlash off
+        </IfModule>
+    </IfModule>
```

Comment:
@kyrofa Do you think it is possible to automatically generate this from the Nextcloud tarball?
Maybe we could use some features in the snapcraft.yaml file or how could this work?

Reply:
I mean I was able to write a script that does that (https://github.com/nextcloud/nextcloud-snap/issues/412#issuecomment-789663913) but is it possible to execute it from the snapcraft.yaml?

Reply:
I think I've found a way how to do this